scripts/tasks.py

```diff
@@ -18,31 +18,42 @@ def make_powder(config):
     setup = config.setup
     task = config.make_powder
     """ Generate the max, avg, and std powders for a given run. """
+    taskdir = os.path.join(setup.root_dir, 'powder')
+    os.makedirs(taskdir, exist_ok=True)
     rd = RunDiagnostics(exp=setup.exp,
                         run=setup.run,
                         det_type=setup.det_type)
-
     logger.debug(f'Computing Powder for run {setup.run} of {setup.exp}...')
-    rd.compute_run_stats(n_images=task.n_images,
-                         powder_only=True)
-    logger.info(f'Saving Powders to {setup.root_dir}')
-    rd.save_powders(setup.root_dir)
+    rd.compute_run_stats(max_events=task.max_events, mask=task.mask)
+    logger.info(f'Saving Powders and plots to {taskdir}')
+    rd.save_powders(taskdir)
+    rd.visualize_powder(output=os.path.join(taskdir, f"powder_r{rd.psi.run:04}.png"))
+    rd.visualize_stats(output=os.path.join(taskdir, f"stats_r{rd.psi.run:04}.png"))
     logger.debug('Done!')
     
 def opt_distance(config):
     from btx.diagnostics.geom_opt import GeomOpt
+    from btx.misc.metrology import modify_crystfel_header, generate_geom_file
     setup = config.setup
     task = config.opt_distance
     """ Optimize the detector distance from an AgBehenate run. """
+    taskdir = os.path.join(setup.root_dir, 'geom')
```

Comment:
Where do we want to document the organization of the task directories? In the wiki, docstring, README?

Reply:
Great point! I think the Wiki would be great (since the structure is enforced through tasks.py, but not if the scripts are run from the command line). I already have an issue open for updating the Wiki, so will make a note of it there.